### PR TITLE
Add repeated_primitive_no_list.parquet

### PR DESCRIPTION
**Pull request by zeevm**
Add file for testing of change in [#6649](https://github.com/apache/arrow-rs/pull/6649)

### CODE REVIEW

File: data/README.md [hunk 1]
```diff
@@ -54,6 +54,7 @@
 | column_chunk_key_value_metadata.parquet | two INT32 columns, one with column chunk key-value metadata {"foo": "bar", "thisiskeywithoutvalue": null} note that the second key "thisiskeywithoutvalue", does not have a value, but the value can be mapped to an empty string "" when read depending on the client |
 | sorting_columns.parquet | INT64 and BYTE_ARRAY columns with first column with nulls first and descending, second column with nulls last and ascending. This file contains two row groups with same data and sorting columns. |
 | old_list_structure.parquet | Single LIST<LIST<INT32>> column with legacy two-level list structure. See [old_list_structure.md](old_list_structure.md) |
+| repeated_primitive_no_list.parquet | INT32 and BYTE_ARRAY REPEATED primitive fields not contained in a LIST annotated group. |
```
Comment (wgtmac) [inline data/README.md:57]: Would you mind adding the following summary of the file, like what we did for `incorrect_map_schema.parquet` and `byte_stream_split.zstd.parquet` above?

```
> parquet-cli meta repeated_no_list.parquet

File path:  repeated_no_list.parquet
Created by: parquet-rs version 53.2.0
Properties: (none)
Schema:
message schema {
  repeated int32 Int32;
  repeated binary String (STRING);
}


Row group 0:  count: 4  65.75 B records  start: 4  total(compressed): 263 B total(uncompressed):263 B
--------------------------------------------------------------------------------
        type      encodings count     avg size   nulls   min / max
Int32   INT32     _ RR_     10        10.90 B    1       "0" / "8"
String  BINARY    _ RR_     10        15.40 B    0       "eight" / "zero"
```

Comment (zeevm) [reply data/README.md]: added

File: data/README.md [hunk 2]
```diff
@@ -430,3 +431,20 @@ message hive_schema {
   }
 }
 ```
+
+## Primitive REPEATED fields with no LIST schema
```
Comment (wgtmac) [inline data/README.md:435]:
```suggestion
## REPEATED primitive fields with no LIST annotation
```

File: data/README.md [hunk 3]
```diff
@@ -54,6 +54,7 @@
 | column_chunk_key_value_metadata.parquet | two INT32 columns, one with column chunk key-value metadata {"foo": "bar", "thisiskeywithoutvalue": null} note that the second key "thisiskeywithoutvalue", does not have a value, but the value can be mapped to an empty string "" when read depending on the client |
 | sorting_columns.parquet | INT64 and BYTE_ARRAY columns with first column with nulls first and descending, second column with nulls last and ascending. This file contains two row groups with same data and sorting columns. |
 | old_list_structure.parquet | Single LIST<LIST<INT32>> column with legacy two-level list structure. See [old_list_structure.md](old_list_structure.md) |
+| repeated_primitive_no_list.parquet | INT32 and BYTE_ARRAY REPEATED primitive fields not contained in a LIST annotated group. See [note](#Primitive-REPEATED-fields-with-no-LIST-schema) |
```
Comment (wgtmac) [inline data/README.md:57]:
```suggestion
| repeated_primitive_no_list.parquet | REPEATED INT32 and BYTE_ARRAY fields without LIST annotation. See [note](#REPEATED-primitive-fields-with-no-LIST-annotation) |
```

**Review by wgtmac — APPROVED** [review]
LGTM. Thanks!

**Comment by zeevm** [comment]
@wgtmac I don't think I can merge the PR correct? Don't have permission.

**Pull request merged**